seq2science/rules/quantification.smk

```diff
@@ -166,22 +166,45 @@ elif config["quantifier"] == "kallistobus":
             {params.options} > {log} 2>&1
             """
 
-    def get_kallistobus_reads(wildcards):
-        reads = []
-        sample = wildcards.sample
+    def get_fastq_pair_reads(wildcards):
+        reads = dict()
         assert sampledict[sample]["layout"] == "PAIRED"
         read_id = get_bustools_rid(config.get("count"))
         #Determine mate for trimming
         if read_id == 0:
-            reads += expand(f"{{trimmed_dir}}/{sample}_R1_trimmed.{{fqsuffix}}.gz", **config)
-            reads += expand("{fastq_dir}/{{sample}}_R2.{fqsuffix}.gz", **config)
+            reads["r1"] = expand("{trimmed_dir}/{{sample}}_{fqext1}_trimmed.{fqsuffix}.gz", **config)
+            reads["r2"] = expand("{fastq_dir}/{{sample}}_{fqext2}.{fqsuffix}.gz", **config)
         elif read_id == 1:
-            reads += expand("{fastq_dir}/{{sample}}_R1.{fqsuffix}.gz", **config)
-            reads += expand(f"{{trimmed_dir}}/{sample}_R2_trimmed.{{fqsuffix}}.gz", **config)
-        else:    
+            reads["r1"] = expand("{fastq_dir}/{{sample}}_{fqext1}.{fqsuffix}.gz", **config)
+            reads["r2"] = expand("{trimmed_dir}/{{sample}}_{fqext2}_trimmed.{fqsuffix}.gz", **config)
+        else:
             raise NotImplementedError
-
         return reads
+        
+    rule fastq_pair:    
+        """
+        Example
+        """
+        input:
+            unpack(get_fastq_pair_reads)
+        output:
+            r1=expand("{fastq_clean_dir}/{{sample}}_clean_{fqext1}.{fqsuffix}.paired.fq", **config),
+            r2=expand("{fastq_clean_dir}/{{sample}}_clean_{fqext2}.{fqsuffix}.paired.fq", **config),
```

Comment:
P.S. we already make the assumption that fqext1 and fqext2 are ALWAYS lexicographically sorted, so they should always be found as R1 > R2.

Reply:
R1 and R2 are used by the rule downstream, so I think this is nice and verbose this way

    input:
         r1=rules.fastq_pair.output.r1,
         r2=rules.fastq_pair.output.r2

Reply:
@siebrenf It can be reduced if I use `fqext` indeed. Since its lexicographically sorted. If I reference it via `rules.fastq_pair_output_reads` in `kallistobus_count` ,It should not cause any troubles I would assume.